Products/CMFPlone/recyclebin.py

```diff
@@ -0,0 +1,584 @@
+            return None
+
+        # Generate a unique ID for the recycled item
+        item_id = str(uuid.uuid4())
+
+        # Generate a meaningful title
+        item_title = "Unknown"
+        if item_type == "CommentTree":
+            # For comment trees, generate a title including the number of comments
+            comment_count = len(obj.get("comments", []))
+            root_comment = None
+
+            # Try to find the root comment to get its text
+            for comment, _ in obj.get("comments", []):
+                if getattr(comment, "comment_id", None) == obj.get("root_comment_id"):
+                    root_comment = comment
+                    break
+
+            # If we found the root comment, get a preview of its text
+            comment_preview = ""
+            if root_comment and hasattr(root_comment, "text"):
+                # Take the first 30 characters of the text as a preview
+                text = getattr(root_comment, "text", "")
+                if text:
+                    if len(text) > 30:
+                        comment_preview = text[:30] + "..."
+                    else:
+                        comment_preview = text
+
+            # Create a meaningful title
+            if comment_preview:
+                item_title = (
+                    f'Comment thread: "{comment_preview}" ({comment_count} comments)'
+                )
+            else:
+                item_title = f"Comment thread ({comment_count} comments)"
+        else:
+            # For regular items, use Title() if available
+            item_title = (
+                obj.Title()
+                if hasattr(obj, "Title")
+                else getattr(obj, "title", "Unknown")
+            )
+
+        # Store metadata about the deletion
+        self.storage[item_id] = {
+            "id": (
+                obj.getId() if hasattr(obj, "getId") else getattr(obj, "id", "unknown")
+            ),
+            "title": item_title,
+            "type": item_type or getattr(obj, "portal_type", "Unknown"),
+            "path": original_path,
+            "parent_path": "/".join(original_container.getPhysicalPath()),
+            "deletion_date": datetime.now(),
+            "size": getattr(obj, "get_size", lambda: 0)(),
+            "object": obj,  # Store the actual object
+        }
+
+        # Check if we need to clean up old items
+        self._check_size_limits()
+
+        return item_id
+
+    def get_items(self):
+        """Return all items in recycle bin"""
+        items = []
+        for item_id, data in self.storage.items():
+            item_data = data.copy()
+            item_data["recycle_id"] = item_id
+            # Don't include the actual object in the listing
+            if "object" in item_data:
+                del item_data["object"]
+            items.append(item_data)
+
+        # Sort by deletion date (newest first)
+        return sorted(items, key=lambda x: x["deletion_date"], reverse=True)
```

Comment:
Copying and sorting at retrieval time could be a bit of a bottleneck if the recycle bin is large. It would be nice to put the items in the recycle bin storage in a structure that can be returned without further processing.